File: jdroid-android/src/main/java/com/jdroid/android/search/SingleLineSuggestionsProvider.java
package com.jdroid.android.search;

import android.annotation.SuppressLint;
import android.content.SearchRecentSuggestionsProvider;

import com.jdroid.android.utils.AppUtils;

/**
 * Declare on your manifest as:
 * 
 * <pre>
 * &lt;provider android:name=".android.common.search.SingleLineSuggestionsProvider"
 * 	android:authorities="[PackageName].SingleLineSuggestionsProvider" />
 * </pre>
 * 
 */
@SuppressLint("Registered")
public class SingleLineSuggestionsProvider extends SearchRecentSuggestionsProvider {
	
	public static final String AUTHORITY = AppUtils.getApplicationId() + ".SingleLineSuggestionsProvider";
	public static final int MODE = DATABASE_MODE_QUERIES;
	
	public SingleLineSuggestionsProvider() {
		setupSuggestions(AUTHORITY, MODE);
	}
}

File: jdroid-android-core/src/main/java/com/jdroid/android/search/SingleLineSuggestionsProvider.kt
package com.jdroid.android.search

import android.annotation.SuppressLint
import android.content.SearchRecentSuggestionsProvider

import com.jdroid.android.utils.AppUtils

/**
 * Declare on your manifest as:
 *
 * <pre>
 * &lt;provider android:name=".android.common.search.SingleLineSuggestionsProvider"
 * android:authorities="[applicationId].SingleLineSuggestionsProvider" />
</pre> *
 */
@SuppressLint("Registered")
class SingleLineSuggestionsProvider : SearchRecentSuggestionsProvider() {

    init {
        setupSuggestions(AUTHORITY, MODE)
    }

    companion object {
        val AUTHORITY = AppUtils.getApplicationId() + ".SingleLineSuggestionsProvider"
        const val MODE = DATABASE_MODE_QUERIES
    }
}